### ISSUE TITLE
Feature proposal: enforcing option to disallow `typeof foo === 'object'`

### ISSUE THREAD

**Issue by jakub-g** [jshint/jshint#2905]
`typeof` has a known quirk of returning `object` for `null` which is rather unfortunate and can lead to bugs. 

It could be interesting to have an option to have JSHint warn on code like `typeof foo === 'object'`

What do you think?

**Comment by mgol** [comment]
How would you detect if an object is, indeed, an object then?

**Comment by jakub-g** [comment]
Yeah it's tricky because in plain JS you would write `typeof x === 'object'
& x !== null` to achieve that. The idea is to have a helper method
`isObject`, or use a library method like `angular.isObject`, and force
everyone to use the that method instead of typeof.

The one particular occurrence of `typeof x === 'object'` in the helper
method can be allowed with a precisely scoped jshintignore.

**Comment by jakub-g** [comment]
Yeah it's tricky because in plain JS you would write `typeof x === 'object' && x !== null` to achieve that. The idea is to have a helper method `isObject`, or use a library method like `angular.isObject`, and force everyone to use the that method instead of typeof.

The one particular occurrence of `typeof x === 'object'` in the helper method can be allowed with a precisely scoped jshintignore.

**Comment by jugglinmike** [comment]
Another workaround would be to obscure the operator's right-hand side, as
described in the latest version of this feature request.

It's true that this could not be enforced consistently, which makes me think
that it may not be worthwhile. But then again, the project has set a precedence
for linting `typeof`'s right-hand side--see [W122 ("Invalid typeof
value")](https://github.com/jshint/jshint/blob/29dd071d66ee16d250ce63c0c4e061b74f7423ac/src/messages.js#L205).
